pkg/scd/dss_report_handler.go

```diff
@@ -0,0 +1,77 @@
+package scd
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/google/uuid"
+	"github.com/interuss/dss/pkg/api"
+	restapi "github.com/interuss/dss/pkg/api/scdv1"
+	dsserr "github.com/interuss/dss/pkg/errors"
+	"github.com/interuss/dss/pkg/logging"
+	"github.com/interuss/stacktrace"
+	"go.uber.org/zap"
+)
+
+// DSSReporting takes care of handling a DSS report.
+type DSSReporting interface {
+	// HandleDssReport handles a DSS report request. Returns the error report passed in 'req' after having set its identifier.
+	HandleDssReport(ctx context.Context, req *restapi.MakeDssReportRequest) (*restapi.ErrorReport, error)
+}
+
+// JSONLoggingDssReportHandler a DSSReportHandler that simply logs the received report as JSON.
+type JSONLoggingDssReportHandler struct {
+	// ReportLogger is the logger to which the received reports will be logged.
+	ReportLogger *zap.Logger
+}
+
+// HandleDssReport logs the received report as a JSON string to a logger.
+func (h *JSONLoggingDssReportHandler) HandleDssReport(ctx context.Context, req *restapi.MakeDssReportRequest) (*restapi.ErrorReport, error) {
+	reportID, err := uuid.NewRandom()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Failed to generate report ID")
+	}
+	rVal := req.Body
+	reportIDStr := reportID.String()
+	rVal.ReportId = &reportIDStr
+	// Serialize the report to a JSON string:
+	jsonReport, err := json.Marshal(req.Body)
+	if err != nil {
+		logging.WithValuesFromContext(ctx, logging.Logger).Error("Failed to serialize DSS Report", zap.Error(err))
+		return nil, stacktrace.PropagateWithCode(err, dsserr.BadRequest, "Failed to serialize DSS Report")
+	}
+	h.ReportLogger.Info("Received DSS Report", zap.String("reportID", reportIDStr), zap.String("report", string(jsonReport)))
```

Comment:
Using `zap.String` for the report will escape parts of the JSON making it not too readable. Here using `zap.Any` should produce better result.

Reply:
Actually I had in mind that the logs are necessarily JSON-encoded. While it's the default this is configurable so we can't assume that. In that case we could encode it in base64, but that may be going too far for now. So let's go ahead with this version.

Reply:
Yes, the logs are in JSON, and in the current setup (to the best of my understanding), we'll have one field of the overall JSON log be raw, escaped JSON. This should be easy to process.